server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1403,6 +1402,23 @@ private void closeConsistencyCheck() {
     // TODO check lastFlushID and lostCompactID - ACCUMULO-1290
   }
 
+  private void compareToDataInMemory(TabletMetadata tabletMetadata) {
+    if (!tabletMetadata.getFilesMap().equals(getDatafileManager().getDatafileSizes())) {
+      String msg = "Data files in " + extent + " differ from in-memory data "
+          + tabletMetadata.getFilesMap() + " " + getDatafileManager().getDatafileSizes();
+      log.error(msg);
+      throw new RuntimeException(msg);
+    }
+  }
+
+  public void compareTabletInfo(Pair<Long,TabletMetadata> tabletInfo) {
+    // if the counter didn't change, compare metadata to what is in memory
+    if (tabletInfo.getFirst() == this.getUpdateCounter()) {
+      this.compareToDataInMemory(tabletInfo.getSecond());
+    }
+    // if counter did change, don't compare metadata and try again later
+  }
```

Comment:
Need to synchronize this method on the tablet lock and check if the tablet is closed.  We gathered a snapshot of all online tablets at a point in time, it possible that some of the tablets in that snapshot are closed when we try to inpsect them.  The reason we need to syncronize on the tablet lock is that we are inspecting multiple tablet instance vars (closed state, update counter, and files map) and we want to consider all of the instance vars atomically. W/o sync the closed state can change after we inspect it, the update counter can change after we inspect it, etc.

```suggestion
  public synchronized void compareTabletInfo(Pair<Long,TabletMetadata> tabletInfo) {
     if(isClosing() || isClosed()) {
          return;
     }
  
    // if the counter didn't change, compare metadata to what is in memory
    if (tabletInfo.getFirst() == this.getUpdateCounter()) {
      this.compareToDataInMemory(tabletInfo.getSecond());
    }
    // if counter did change, don't compare metadata and try again later
  }
```